sbol_utilities/helper_functions.py

```diff
@@ -0,0 +1,71 @@
+import sbol3
+import filecmp
+import difflib
+
+#########################
+# Collection of shared helper functions for utilities package
+
+
+# Flatten list of lists into a single list
+def flatten(collection):
+    return [item for sublist in collection for item in sublist]
+
+
+#########################
+# Kludge materials that should be removed after certain issues are resolved
+
+
+# Workaround for pySBOL3 issue #231: should be applied to every iteration on a collection of SBOL objects
+def id_sort(i: iter):
+    sortable = list(i)
+    sortable.sort(key=lambda x: x.identity if isinstance(x, sbol3.Identified) else x.lookup().identity)
```

Comment:
The `else` clause seems kind of circuitous.  Wouldn't the following accomplish the same thing?
```
sortable.sort(key=lambda x: x.identity if isinstance(x, sbol3.Identified) else x)
```

Reply:
I don't think it will, because in that case x is not a string but a ReferencedURI

Reply:
ReferencedURI should return a string

Reply:
Just tested it, and it looks like you are correct.